js/views/menuView.js

```diff
@@ -0,0 +1,97 @@
+import elements from "./base";
+
+/**
+ * @typedef {import('../../utils/markUp.js')}
+ */
+import {
+  articleLiMarkUp,
+  columnMarkUp,
+  subArticleMarkUp,
+  menuMarkUp,
+  navBarSocialMarkUp,
+  navBarMenuMarkUp
+} from "../../utils/markUp";
+
+const uniqid = require("uniqid");
+
+/**
+ *
+ * @param { string } name
+ * @param { object } subMenu
+ * @param { string } type
+ * @param { string } id An string that each subMenu and they submenu articles will have to identified witch subArticle belongs to each submenu.
+ * this id will be assigned when the subArticle and the submenu is created.
+ *
+ * @returns { string } Returns an html snippet with all the menus that will be inserted in the html.
+ */
+const createMenu = ({ name, subMenu, type, id }) => {
+  let menu = "";
```

Comment:
Why is menu, subArticle and column are `let` instead of `const`?
You don't need to use `let` if you are no reassigning the value. The same for `article` bellow for example.